### PR TITLE
CI optimization

### PR DESCRIPTION
**Pull request by gaborbernat**
1. Run all tests on release commit.
2. Do not run pytest if the source files do not change.
3. Remove Makefile for Sphinx.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -61,17 +60,25 @@ matrix:
...
 - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then source .travis-osx; fi
+- export BUILD="setup.py pyproject.toml MANIFEST.in setup.cfg tox.ini .travis.yml appveyor.yml"
+- export SRC=$(git diff --name-only $TRAVIS_COMMIT_RANGE -- '*.py' $BUILD | grep -v conf.py | grep -v "tasks/.*.py")
+- [[ "$TOXENV"!="py" || (-n "$SRC" && "$TOXENV"="py")]] && export RUN=1
 
 install:
-- pip install -U six
-- pip install --pre -U tox
-- tox --notest
+- [ -n "$RUN" ] && pip install --pre -U tox
```
Comment (obestwalter) [inline .travis.yml]: `[ -n "$RUN" ]` - never saw this before what does this do? Can you point me to some documentation? Would be interested to learn more about these options.

File: doc/Makefile [hunk 2]
```diff
@@ -1,130 +0,0 @@
-# Makefile for Sphinx documentation
```
Comment (obestwalter) [inline doc/Makefile]: Good call - nobody used this anymore for a long time, I guess.

File: .travis.yml (hunk 1 above)
Comment (gaborbernat) [reply .travis.yml]: http://wiki.bash-hackers.org/commands/classictest#string_tests

Comment (obestwalter) [reply .travis.yml]: thx

**Pull request merged**